neal/sampler.py

```diff
@@ -97,75 +99,95 @@ def __init__(self):
         self.parameters = {'beta_range': [],
                            'num_reads': [],
                            'num_sweeps': [],
+                           'num_sweeps_per_beta': [],
                            'beta_schedule_type': ['beta_schedule_options'],
                            'seed': [],
                            'interrupt_function': [],
                            'initial_states': [],
                            'initial_states_generator': [],
                            }
-        self.properties = {'beta_schedule_options': ('linear', 'geometric')
-                           }
+        self.properties = {'beta_schedule_options': ('linear', 'geometric',
+                                                     'custom')}
 
-    def sample(self, bqm, beta_range=None, num_reads=None, num_sweeps=1000,
-               beta_schedule_type="geometric", seed=None, interrupt_function=None,
-               initial_states=None, initial_states_generator="random", **kwargs):
-        """Sample from a binary quadratic model using an implemented sample method.
+    def sample(self, bqm, beta_range=None, num_reads=None, num_sweeps = None,
+               num_sweeps_per_beta=1, beta_schedule_type="geometric", seed=None,
+               interrupt_function=None, beta_schedule = None,
+               initial_states=None, initial_states_generator="random",
+               **kwargs):
+        """Sample from a binary quadratic model using an implemented sample 
+        method.
 
         Args:
             bqm (:class:`dimod.BinaryQuadraticModel`):
                 The binary quadratic model to be sampled.
 
-            beta_range (tuple, optional):
+            beta_range (tuple, optional, default = None):
```

Comment:
"tuple" --> also accepts a list

Reply:
Yeah - this was an issue I noticed. The example code provides a list, but the argument specification is tuple. I suppose I should just write (tuple, -> (tuple or list,